internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -170,6 +176,40 @@ func (r *AgentServiceConfigReconciler) Reconcile(origCtx context.Context, req ct
 		return reconcile.Result{}, nil
 	}
 
+	if instance.DeletionTimestamp.IsZero() {
+		if !funk.ContainsString(instance.GetFinalizers(), agentServiceConfigFinalizerName) {
```

Comment:
Consider also [using controllerutil](https://github.com/kubernetes-sigs/controller-runtime/blob/b8db76e/pkg/controller/controllerutil/controllerutil.go#L375) for this.